### PR TITLE
Synchronous Custom Elements Polyfill

### PR DESCRIPTION
**Pull request by jridgewell**
`appendChild` and friends behave in a weird psuedo-synchronous way:

```js
class XDiv extends HTMLElement {
  connectedCallback() {
    console.log('connected');
  }
  disconnectedCallback() {
    console.log('disconnected');
  }
}
customElements.define('x-div', XDiv);

console.log('before append');
document.body.appendChild(document.createElement('x-div'));
console.log('after append');

console.log('before remove');
document.body.removeChild(document.body.lastElementChild);
console.log('after remove');
```

This logs:

1. 'before append'
2. 'connected'
3. 'after append'
4. 'before remove'
5. 'disconnected'
6. 'after remove'

This means that upgrade/`connectedCallback` happens before `appendChild` completes, and the similar for `removeChild`/etc. So, we need to support that in our polyfill.

### CODE REVIEW

**Comment by jridgewell** [comment]
[Bumped](https://github.com/ampproject/amphtml-build-artifacts/commit/d66680c070881ad7c0c650f3a28525a86d88f42e) max-size.

**Comment by jridgewell** [comment]
Ping @choumx

File: src/polyfills/custom-elements.js [hunk 1]
```diff
@@ -455,15 +463,28 @@ class Registry {
     this.query_ = name;
 
     // The first registered name starts the mutation observer.
-    const observer = new this.win_.MutationObserver(records => {
+    this.mutationObserver_ = new this.win_.MutationObserver(records => {
```
Comment (choumx) [inline src/polyfills/custom-elements.js]: When will the callback function be invoked now that we have `sync()`?

Comment (jridgewell) [reply src/polyfills/custom-elements.js]: Hopefully never, it's there just in case I've screwed up.

**Pull request merged**